src/bin-utils/get-script-by-prefix.test.js

```diff
@@ -0,0 +1,224 @@
+  const actual = scriptToObject('foo', 'hello')
+  expect(isPlainObject(actual)).toBeTruthy()
+  expect(isNull(actual)).toBeFalsy()
+})
+
+test('scriptToObject returns non-null object when script is string ', () => {
+  const scriptArg = {
+    script: 'hello',
+  }
+  const actual = scriptToObject('foo', scriptArg)
+  expect(isPlainObject(actual)).toBeTruthy()
+  expect(isNull(actual)).toBeFalsy()
+})
+
+test('getScriptByPrefix returns null when no matching script', () => {
+  const config = {scripts: {}}
+  const message = getScriptByPrefix(config, 'w')
+  expect(isNull(message)).toBeTruthy()
+})
+
+test('getScriptByPrefix returns null when no matching script', () => {
+  const config = {scripts: {}}
+  const message = getScriptByPrefix(config, 'w.a.b')
+  expect(isNull(message)).toBeTruthy()
+})
+
+test('getScriptByPrefix resolves single prefixes with string values', () => {
+  const config = {
+    scripts: {
+      watch: 'echo watch',
+    },
+  }
+  const actual = getScriptByPrefix(config, 'w')
+  const expected = {
+    script: 'echo watch',
+    name: 'watch',
+    description: '',
+  }
+  expect(actual).toEqual(expected)
+})
+
+test('getScriptByPrefix resolves nested prefixes with string values', () => {
+  const config = {
+    scripts: {
+      watch: {
+        js: {
+          foo: 'echo foo',
+        },
+      },
+    },
+  }
+  const actual = getScriptByPrefix(config, 'w.j.f')
+  const expected = {
+    script: 'echo foo',
+    name: 'watch.js.foo',
+    description: '',
+  }
+  expect(actual).toEqual(expected)
+})
+
+test(
+  'getScriptByPrefix resolves single prefixes with object values && script key',
+  () => {
+    const config = {
+      scripts: {
+        foo: {
+          description: 'Foo script',
+          script: 'echo foo',
+        },
+      },
+    }
+    const actual = getScriptByPrefix(config, 'f')
+    const expected = {
+      script: 'echo foo',
+      name: 'foo',
+      description: 'Foo script',
+    }
+    expect(actual).toEqual(expected)
+  },
+)
+
+test(
+  'getScriptByPrefix resolves nested prefixes with object values && script key',
+  () => {
+    const config = {
+      scripts: {
+        watch: {
+          foo: {
+            description: 'Foo script',
+            script: 'echo foo',
+          },
+        },
+      },
+    }
+    const actual = getScriptByPrefix(config, 'w.f')
+    const expected = {
+      script: 'echo foo',
+      name: 'watch.foo',
+      description: 'Foo script',
+    }
+    expect(actual).toEqual(expected)
+  },
+)
+
+test('getScriptByPrefix resolves ambiguities with single prefix', () => {
+  const config = {
+    scripts: {
+      foo: {
+        js: {
+          default: 'foo.js script',
+          script: 'echo foo.js',
+        },
+      },
+      foobar: {
+        description: 'Foobar script',
+        script: 'echo foobar',
+      },
+    },
+  }
+  const actual = getScriptByPrefix(config, 'f')
```

Comment:
This is correct. It currently finds the first match. Though, just to be clear. Currently if you provided `f` with these scripts, it would not match `foo.js`. You have to provide at least one character for each level of nesting. So this test is resolving correctly. :+1: